llvm/include/llvm/CodeGen/RegisterUsageInfo.h

```diff
@@ -66,6 +63,52 @@ class PhysicalRegisterUsageInfo : public ImmutablePass {
   const LLVMTargetMachine *TM = nullptr;
 };
 
+class PhysicalRegisterUsageInfoWrapperLegacy : public ImmutablePass {
+  std::unique_ptr<PhysicalRegisterUsageInfo> PRUI;
```

Comment:
I don't know much about regmasks, but do you mean forego setting it in the call instruction MO?

Reply:
No. I mean this analysis largely serves just as a place to hold allocated regmasks. There is a parallel mechanism for storing custom register masks in MachineRegisterInfo already. We need to consolidate these, and improve the serialization for them (i.e. introduce some kind of printed MachineFunction section for contained regmasks with an ID that the instructions can refer to them by inline, otherwise we'll have hundreds of lines printed for every call instruction)

Reply:
From what I remember this analysis is basically equivalent to `MachineRegisterInfo::UsedPhysRegMask` however I think for interprocedural regalloc we did not want to keep the full `MachineRegisterInfo` around for every function in the module, and that's why this simpler analysis pass exists? (and I think this analysis is not used anyway when IPRA is not enabled)

Reply:
The problem is then you can't properly serialize the MIR. If your testcase depends on the IPRA state, -start-before to run the RA pipeline does not work